core/scheduler/src/main/scala/org/apache/openwhisk/core/scheduler/container/ContainerManager.scala

```diff
@@ -248,31 +250,6 @@ class ContainerManager(jobManagerFactory: ActorRefFactory => ActorRef,
     ContainerKeyMeta(revision, invokerId, containerId)
   }
 
-  // Filter out messages which can use warmed container
-  private def filterWarmedCreations(msgs: List[ContainerCreationMessage]) = {
-    msgs.filter { msg =>
-      val warmedPrefix =
-        containerPrefix(ContainerKeys.warmedPrefix, msg.invocationNamespace, msg.action, Some(msg.revision))
-      val chosenInvoker = warmedContainers
-        .filter(!inProgressWarmedContainers.values.toSeq.contains(_))
-        .find { container =>
-          if (container.startsWith(warmedPrefix)) {
-            logging.info(this, s"Choose a warmed container $container")
-            inProgressWarmedContainers.update(msg.creationId.asString, container)
-            true
-          } else
-            false
-        }
-        .map(_.split("/").takeRight(3).apply(0))
-      if (chosenInvoker.nonEmpty) {
-        creationJobManager ! RegisterCreationJob(msg)
-        sendCreationContainerToInvoker(messagingProducer, chosenInvoker.get.toInt, msg)
```

Comment:
There were side effects in this method like this one.
This method is supposed to filter warmed creations but it also sent creation messages to invokers.
This kind of side effect makes it harder to test this method, I removed side effects.